plugins/modules/nutanix_vm.py

```diff
@@ -385,69 +821,146 @@ def entry_point(module, client):
     return func(module.params, client)
 
 
-def create_vm_spec(params, vm_spec):
+def create_vm_spec(params, vm_spec, client):
```

Comment:
Please add doc string for function which will explain about purpose, in args, return value etc